testnet/stacks-node/src/tests/neon_integrations.rs

```diff
@@ -7387,3 +7894,2158 @@ fn test_chainwork_partial_interval() {
+
+    // only tx_edge should be in the mempool
+    assert!(get_unconfirmed_tx(&http_origin, &tx_edge_txid).is_some());
+    assert!(get_unconfirmed_tx(&http_origin, &tx_exceeds_txid).is_none());
+    assert!(get_unconfirmed_tx(&http_origin, &tx_high_txid).is_none());
+
+    channel.stop_chains_coordinator();
+}
+
+fn find_new_files(dirp: &str, prev_files: &HashSet<String>) -> (Vec<String>, HashSet<String>) {
+    let dirpp = Path::new(dirp);
+    debug!("readdir {}", dirp);
+    let cur_files = fs::read_dir(dirp).unwrap();
+    let mut new_files = vec![];
+    let mut cur_files_set = HashSet::new();
+    for cur_file in cur_files.into_iter() {
+        let cur_file = cur_file.unwrap();
+        let cur_file_fullpath = dirpp.join(cur_file.path()).to_str().unwrap().to_string();
+        test_debug!("file in {}: {}", dirp, &cur_file_fullpath);
+        cur_files_set.insert(cur_file_fullpath.clone());
+        if prev_files.contains(&cur_file_fullpath) {
+            test_debug!("already contains {}", &cur_file_fullpath);
+            continue;
+        }
+        test_debug!("new file {}", &cur_file_fullpath);
+        new_files.push(cur_file_fullpath);
+    }
+    debug!(
+        "Checked {} for new files; found {} (all: {})",
+        dirp,
+        new_files.len(),
+        cur_files_set.len()
+    );
+    (new_files, cur_files_set)
+}
+
+fn spawn_follower_node(
+    initial_conf: &Config,
+) -> (
+    Config,
+    neon::RunLoopCounter,
+    PoxSyncWatchdogComms,
+    CoordinatorChannels,
+) {
+    let bootstrap_node_public_key = {
+        let keychain = Keychain::default(initial_conf.node.seed.clone());
+        let mut pk = keychain.generate_op_signer().get_public_key();
+        pk.set_compressed(true);
+        pk.to_hex()
+    };
+
+    let (mut conf, _) = neon_integration_test_conf();
+    conf.node.set_bootstrap_nodes(
+        format!(
+            "{}@{}",
+            &bootstrap_node_public_key, initial_conf.node.p2p_bind
+        ),
+        conf.burnchain.chain_id,
+        conf.burnchain.peer_version,
+    );
+
+    conf.events_observers.push(EventObserverConfig {
+        endpoint: format!("localhost:{}", test_observer::EVENT_OBSERVER_PORT),
+        events_keys: vec![EventKeyType::AnyEvent],
+    });
+
+    conf.initial_balances = initial_conf.initial_balances.clone();
+    conf.burnchain.epochs = initial_conf.burnchain.epochs.clone();
+    conf.burnchain.ast_precheck_size_height =
+        initial_conf.burnchain.ast_precheck_size_height.clone();
+
+    let mut run_loop = neon::RunLoop::new(conf.clone());
+    let blocks_processed = run_loop.get_blocks_processed_arc();
+    let channel = run_loop.get_coordinator_channel().unwrap();
+    let pox_sync = run_loop.get_pox_sync_comms();
+
+    thread::spawn(move || run_loop.start(None, 0));
+
+    // Give the run loop some time to start up!
+    wait_for_runloop(&blocks_processed);
+
+    (conf, blocks_processed, pox_sync, channel)
+}
+
+// TODO: test in epoch 2.1 with parser_v2
```

Comment:
Should we do these todo items in this PR?

Reply:
I have a separate issue for them here: #3368